manual_run/weight_run.py

```diff
@@ -0,0 +1,127 @@
+import csv
+import os
+from math import ceil
+from os import path
+
+import typer
+
+from api.ai.interfaces.algorithm_config import PriorityAlgorithmConfig
+from api.dataclasses.enums import AlgorithmType
+from benchmarking.evaluations.metrics.cosine_similarity import \
+    AverageCosineDifference
+from benchmarking.runs.interfaces import Run
+from benchmarking.simulation.insight import Insight
+from benchmarking.simulation.simulation_set import (SimulationSet,
+                                                    SimulationSetArtifact)
+from benchmarking.simulation.simulation_settings import SimulationSettings
+from manual_run.attributes import Attributes
+from manual_run.data_provider import DataProvider
+from manual_run.default_scenario import DefaultScenario
+from manual_run.map_columns import MapColumns
+from manual_run.variables import Variables
+
+
+class WeightRun(Run):
+    def start(self, num_trials: int = 1, generate_graphs: bool = True):
+        scenario = DefaultScenario()
+
+        # Metrics to be used for evaluation
+        metrics = [
+            AverageCosineDifference(
+                name="Score Cosine Difference",
+                attribute_filter=[Attributes.SCORE.value],
+            ),
+            AverageCosineDifference(
+                name="Timeslot Cosine Difference",
+                attribute_filter=[Attributes.TIMESLOT_AVAILABILITY.value],
+            ),
+        ]
+        student_provider = DataProvider()
+        artifact: SimulationSetArtifact = SimulationSet(
+            settings=SimulationSettings(
+                num_teams=ceil(student_provider.num_students / Variables.team_size),
+                scenario=scenario,
+                student_provider=student_provider,
+                cache_key=f"csv_weight_run/weight_run/",
+            ),
```

Comment:
If I remove this then the results don't print.

Reply:
What do you mean? Like if you remove the cache key? Or the entire SimulationSet creation?

Reply:
Sorry ignore this. I had an error which is now fixed I didn't mean to publish that comment